### PR TITLE
Add support for auto-managing dependencies using Pyup.io w/Pipfile

### PR DESCRIPTION
**Pull request by stephendonner**
@chartjes @davehunt thoughts?

Passing ad-hoc here: https://qa-preprod-master.fxtest.jenkins.stage.mozaws.net/job/kinto.adhoc/23/console

This will also require, of course, that someone with Admin privs on the Kinto account enable the corresponding @pyup-bot support.

Note: I didn't add ```pipenv check --style .``` to the .travis.yml file, due to the following issues it output:

```
Stephens-MacBook-Pro:kinto-integration-tests stephendonner$ pipenv check --style .
/Users/stephendonner/kinto-integration-tests/./conftest.py:0:1: I002 no configuration found (.isort.cfg or [isort] in configs)
/Users/stephendonner/kinto-integration-tests/./config-tests/test_kinto_signer.py:0:1: I002 no configuration found (.isort.cfg or [isort] in configs)
/Users/stephendonner/kinto-integration-tests/./config-tests/test_kintowe_crud.py:0:1: I002 no configuration found (.isort.cfg or [isort] in configs)
/Users/stephendonner/kinto-integration-tests/./config-tests/test_server_details.py:51:5: E999 SyntaxError: invalid syntax
/Users/stephendonner/kinto-integration-tests/./config-tests/test_dlc.py:0:1: I002 no configuration found (.isort.cfg or [isort] in configs)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,15 +17,28 @@ To run the tests, you need to have the following installed:
...
-You can run these tests using the following commands from inside the root directory for the project.
+You can run these tests using the following commands from inside the root directory for the project:
 
 ```shell
-pipenv install
-pipenv shell
-pytest -m TEST_TYPE --env=TEST_ENV
+$ pipenv install
+$ pipenv shell
+$ pytest -m TEST_TYPE --env=TEST_ENV
```
Comment (davehunt) [inline README.md]: The two final lines can be replaced with this single one:

```
pipenv run pytest -m TEST_TYPE --env TEST_ENV
```

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM python:alpine
 RUN apk add --update make gcc python3-dev musl-dev libffi-dev openssl openssl-dev
 WORKDIR /app
 RUN pip install pipenv
-COPY Pipfile Pipfile.lock /app/
-RUN pipenv install --deploy --system
+COPY Pipfile /app/
+RUN pipenv install --system --skip-lock
```
Comment (davehunt) [inline Dockerfile]: This means we're installing the latest versions of all dependencies whenever we build the docker image. Unless we pin all dependencies in the Pipfile, then I don't think this is a good idea.

File: README.md [hunk 3]
```diff
@@ -17,15 +17,28 @@ To run the tests, you need to have the following installed:
...
 ```
 
 * `TEST_TYPE` is `dist` for `kinto-dist` deployments, `settings` for `kinto-settings` deployments and `webextensions` for `kintowe` deployments
 * `TEST_ENV` is one of the environments listed in the `manifest.ini` file.
+
+### Running the tests using Docker ###
```
Comment (davehunt) [inline README.md]: Nit: No need for the trailing ###

File: Dockerfile (hunk 2 above)
Comment (stephendonner) [reply Dockerfile]: Great point; originally, I wasn't intending on doing so, but it just hadn't crossed my mind.  @chartjes?

File: README.md (hunk 3 above)
Comment (stephendonner) [reply README.md]: Fixed; can't believe a) I did that b) didn't spot it, later - thx

File: .travis.yml [hunk 4]
```diff
@@ -2,8 +2,11 @@ language: python
 python:
   - 3.6
 install:
+  - pip install -r pipenv.txt
   - pip install flake8
```
Comment (davehunt) [inline .travis.yml]: We can remove this line in a followup commit, as we're now running `flake8` via `pipenv`

File: pipenv.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+pipenv==10.1.0
```
Comment (davehunt) [inline pipenv.txt]: We should add this to `pyup.yml` in a followup commit, so we get PRs when pipenv is updated.

**Pull request merged**